lbrynet/metadata/Metadata.py

```diff
@@ -14,73 +14,27 @@ def verify_name_characters(name):
     return True
 
 
-def validate_sources(sources):
-    for source in sources:
-        assert source in SOURCE_TYPES, "Unknown source type: %s" % str(source)
-    return True
-
-
-class Metadata(Validator):
-    MV001 = "0.0.1"
-    MV002 = "0.0.2"
-    MV003 = "0.0.3"
-    CURRENT_METADATA_VERSION = MV003
-
-    METADATA_REVISIONS = {}
-
-    METADATA_REVISIONS[MV001] = [
-        (Validator.REQUIRE, 'title', skip_validate),
-        (Validator.REQUIRE, 'description', skip_validate),
-        (Validator.REQUIRE, 'author', skip_validate),
-        (Validator.REQUIRE, 'language', skip_validate),
-        (Validator.REQUIRE, 'license', skip_validate),
-        (Validator.REQUIRE, 'content-type', skip_validate),
-        (Validator.REQUIRE, 'sources', validate_sources),
-        (Validator.OPTIONAL, 'thumbnail', skip_validate),
-        (Validator.OPTIONAL, 'preview', skip_validate),
-        (Validator.OPTIONAL, 'fee', verify_supported_currency),
-        (Validator.OPTIONAL, 'contact', skip_validate),
-        (Validator.OPTIONAL, 'pubkey', skip_validate),
-    ]
-
-    METADATA_REVISIONS[MV002] = [
-        (Validator.REQUIRE, 'nsfw', skip_validate),
-        (Validator.REQUIRE, 'ver', skip_validate),
-        (Validator.OPTIONAL, 'license_url', skip_validate),
-    ]
+class Metadata(StructuredDict):
+    current_version = '0.0.3'
 
-    METADATA_REVISIONS[MV003] = [
-        (Validator.REQUIRE, 'content_type', skip_validate),
-        (Validator.SKIP, 'content-type'),
-        (Validator.OPTIONAL, 'sig', skip_validate),
-        (Validator.IF_KEY, 'sig', (Validator.REQUIRE, 'pubkey', skip_validate), Validator.DO_NOTHING),
-        (Validator.IF_KEY, 'pubkey', (Validator.REQUIRE, 'sig', skip_validate), Validator.DO_NOTHING),
-    ]
+    def __init__(self, metadata, migrate=True, target_version=None):
+        self._versions = [
```

Comment:
this would be better as a class variable not an instance variable.

It does mean that `_migrate_001_to_002`, etc will need to be static methods that take a dictionary as an argument and modify it.